commands/deploy.go

```diff
@@ -261,7 +261,28 @@ Error: %s`, fprocessErr.Error())
 				function.ReadOnlyRootFilesystem = deployFlags.readOnlyRootFilesystem
 			}
 
-			statusCode := proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, allAnnotations, functionResourceRequest1, function.ReadOnlyRootFilesystem, tlsInsecure)
+			deploySpec := &proxy.DeployFunctionSpec{
+				FProcess:                function.FProcess,
+				Gateway:                 services.Provider.GatewayURL,
+				FunctionName:            function.Name,
+				Image:                   function.Image,
+				RegistryAuth:            function.RegistryAuth,
+				Language:                function.Language,
+				Replace:                 deployFlags.replace,
+				EnvVars:                 allEnvironment,
+				Network:                 services.Provider.Network,
+				Constraints:             functionConstraints,
+				Update:                  deployFlags.update,
+				Secrets:                 deployFlags.secrets,
+				Labels:                  allLabels,
+				Annotations:             allAnnotations,
+				FunctionResourceRequest: functionResourceRequest1,
```

Comment:
I am not sure `functionResourceRequest1` why `1` is appended to the name. Maybe we can have a better name for this variable?

Reply:
i'll change that